src/components/chart/state-history-chart-timeline.ts

```diff
@@ -143,11 +143,17 @@ export class StateHistoryChartTimeline extends LitElement {
             }
           },
           afterUpdate: (y) => {
+            const yWidth = this.showNames
+              ? y.width
+              : computeRTL(this.hass)
+              ? 0
+              : y.left;
             if (
-              this._yWidth !== Math.floor(y.width) &&
+              yWidth != null &&
```

Comment:
Why `yWidth` can be `undefined` ?

Reply:
Isn't this enough ?
```ts
const yWidth = this.showNames ? y.width : 0
```

Reply:
Unfortunately I don't think so. 

I tried this at first, but it seems chart.js when there is no label, adds some pixels to left padding inside the canvas, before the bar starts. 

If there is a label, the label name length is captured by y.width, and y.left is 0. 
When there is no label, y.width is 0 and y.left is something like ~18 pixels.

Reply:
> Why yWidth can be undefined ?

It's not typically, but during initialization I watched this function and it goes through a few rounds with y.width being undefined, or y.left being undefined, I just thought it safer to ignore these and not send event for those cases.

Reply:
I think it's safer to fallback to 0 in that case to avoid checking undefined in the `state-history-chart.ts`.

```ts
const yWidth = this.showNames
    ? (y.width ?? 0)
    : computeRTL(this.hass)
    ? 0
    : (y.left ?? 0);
```

Reply:
The reason why I check for undefined in state-history-charts was because the default `this._yWidth` is 0. If this function computes a `yWidth` of 0, that cause it to not fire the `y-width-changed` event. 

Therefore if the first chart doesn't fire the changed event, state-history-charts will have `_childYWidths` array of something like `[empty, 50]`. Without checking for null, the Math.max of this is `NaN`, but in reality I want the Math.max to treat empty as 0.

Reply:
Oh I see the issue... because we use an array... We must use an object and do :
```ts
this._maxYWidth = Math.max(Object.values(this._childYWidths));
```

Reply:
Updated as you suggested.